### PR TITLE
chore(ci): set PR E2E tests timeout to 30 min

### PR DESCRIPTION
**Pull request by yakimant**

### CODE REVIEW

File: ci/Jenkinsfile.tests-e2e [hunk 1]
```diff
@@ -249,3 +249,11 @@ def getDefaultTestScopeFlag() {
     return '-m=critical'
   }
 }
+
+def getTestStageTimeout() {
+  if (JOB_NAME == "status-desktop/systems/linux/x86_64/tests-e2e") {
```
Comment (anastasiyaig) [inline ci/Jenkinsfile.tests-e2e:254]: arent we using the same job everywhere?  How we understand which is nightly run, PR run or manually started job?

Comment (yakimant) [reply ci/Jenkinsfile.tests-e2e:254]: I copy-pasted it from `getDefaultTestScopeFlag()`, but I can also rely on it, so if test scope flag is `-m=critical` - set timeout to 30 min.

**Review by anastasiyaig — APPROVED** [review]

**Comment by anastasiyaig** [comment]
@yakimant please rebase to recent master so tests can pass here, this branch is missing one commit important for the tests

File: ci/Jenkinsfile.tests-e2e [hunk 2]
```diff
@@ -249,3 +249,11 @@ def getDefaultTestScopeFlag() {
...
   }
 }
+
+def getTestStageTimeout() {
+  if (TEST_SCOPE_FLAG == '-m=critical') {
+    return 30
+  } else {
+    return 120
+  }
+}
```
Comment (jakubgs) [inline ci/Jenkinsfile.tests-e2e:259]: This could be one line:
```suggestion
def getTestStageTimeout() {TEST_SCOPE_FLAG == '-m=critical' ? 30 : 120 }
```

**Review by jakubgs — APPROVED** [review]

**Comment by yakimant** [comment]
Timeout works:
https://ci.status.im/job/status-desktop/job/e2e/job/prs/511/

**Pull request merged**